app/src/main/java/com/stevesoltys/seedvault/transport/backup/PackageService.kt

```diff
@@ -50,17 +53,14 @@ internal class PackageService(
                 }
             }
 
-            val eligibleApps =
-                backupManager.filterAppsEligibleForBackupForUser(myUserId, packages.toTypedArray())
-
-            // log eligible packages
-            if (Log.isLoggable(TAG, INFO)) {
-                Log.i(TAG, "Filtering left ${eligibleApps.size} eligible packages:")
-                logPackages(eligibleApps.toList())
-            }
+//            val eligiblePackages = packages.filter { packageInfo ->
+//                isAppEligibleForBackup(packageInfo,
+//                    if (packageInfo.applicationInfo.backupAgentName != null)
+//                        pkg.applicationInfo.flags and FLAG_FULL_BACKUP_ONLY != 0 else true)
+//            }
```

Comment:
Not sure if necessary anymore given the filtering applied in BackupCoordinator's `isAppEligibleForBackup`

Reply:
When testing this PR, an exclusion for DAVx5 did not take effect for me. I reverted this section to the previous eligibleApps code (also bringing back backupManager to the initializer, etc.), and the exclusion took effect. DAVx5 was the only app I attempted to exclude, but I would guess that other apps would have been affected as well.

Reply:
Right, I had disabled this code due to it removing packages that are eligible under D2D. I am not sure if the handling for not backing up the app handling the backup, should be done here or if it can/should be done in `isAppEligibleForBackup`. I would defer to @grote on this

Reply:
These changes, derived from your commented code, have been working for me currently. The backup-handling app seems to be successfully addressed elsewhere. https://gitlab.com/tmw_calyxos_fork/platform_packages_apps_Seedvault/-/commit/96e049b80a62fbd2bcd50d3b6b772ba169c087d0